domain/src/main/kotlin/no/nav/su/se/bakover/domain/revurdering/IdentifiserSaksbehandlingsutfallSomIkkeStøttes.kt

```diff
@@ -0,0 +1,79 @@
+                if (opphørVedRevurdering.grunn.count() > 1) {
+                    utfall.add(RevurderingsutfallSomIkkeStøttes.OpphørAvFlereVilkår)
+                }
+                if (!opphørVedRevurdering.opphørsdatoErTidligesteDatoIRevurdering()) {
+                    utfall.add(RevurderingsutfallSomIkkeStøttes.OpphørErIkkeFraFørsteMåned)
+                }
+                if (setOf(Opphørsgrunn.SU_UNDER_MINSTEGRENSE).containsAll(opphørVedRevurdering.grunn)) {
+                    if (harAndreBeløpsendringerEnnMånederUnderMinstegrense()) {
+                        utfall.add(RevurderingsutfallSomIkkeStøttes.OpphørOgAndreEndringerIKombinasjon)
+                    }
+                    if (!fullstendigOpphør()) {
+                        utfall.add(RevurderingsutfallSomIkkeStøttes.DelvisOpphør)
+                    }
+                }
+                if (setOf(Opphørsgrunn.FOR_HØY_INNTEKT).containsAll(opphørVedRevurdering.grunn)) {
+                    if (harAndreBeløpsendringerEnnMånederMedBeløp0()) {
+                        utfall.add(RevurderingsutfallSomIkkeStøttes.OpphørOgAndreEndringerIKombinasjon)
+                    }
+                    if (!fullstendigOpphør()) {
+                        utfall.add(RevurderingsutfallSomIkkeStøttes.DelvisOpphør)
+                    }
+                }
+                if (setOf(Opphørsgrunn.UFØRHET).containsAll(opphørVedRevurdering.grunn) && harBeløpsendringer(nyBeregning.getMånedsberegninger())) {
+                    utfall.add(RevurderingsutfallSomIkkeStøttes.OpphørOgAndreEndringerIKombinasjon)
+                }
+                if (utfall.isEmpty()) Unit.right() else utfall.left()
+            }
+            OpphørVedRevurdering.Nei -> Unit.right()
+        }
+    }
+
+    private fun fullstendigOpphør(): Boolean {
+        return nyBeregning.getMånedsberegninger().all { it.erSumYtelseUnderMinstebeløp() } || nyBeregning.getMånedsberegninger().all { it.getSumYtelse() == 0 }
+    }
+
+    private fun harAndreBeløpsendringerEnnMånederUnderMinstegrense(): Boolean {
+        return harBeløpsendringer(nyBeregning.getMånedsberegninger().filterNot { it.erSumYtelseUnderMinstebeløp() })
+    }
+
+    private fun harAndreBeløpsendringerEnnMånederMedBeløp0(): Boolean {
+        return harBeløpsendringer(nyBeregning.getMånedsberegninger().filterNot { !it.erSumYtelseUnderMinstebeløp() && it.getSumYtelse() == 0 })
+    }
+
+    private fun harBeløpsendringer(nyeMånedsberegninger: List<Månedsberegning>): Boolean {
+        return tidligereBeregning.getMånedsberegninger().associate { it.periode to it.getSumYtelse() }.let { tidligereMånederOgBeløp ->
+            nyeMånedsberegninger.any { tidligereMånederOgBeløp[it.periode] != it.getSumYtelse() }
+        }
+    }
+
+    private fun OpphørVedRevurdering.Ja.opphørsdatoErTidligesteDatoIRevurdering(): Boolean {
+        return this.opphørsdato == nyBeregning.getMånedsberegninger().minByOrNull { it.periode.fraOgMed }!!.periode.fraOgMed
```

Comment:
Gir vel ikke mening å ha Beregning med månedsberegninger som tom liste? Kan bytte til NEL en gang?

Reply:
Det er definitivt et bra case for nel, enig i det.